components/status_icon.tsx

```diff
@@ -36,10 +35,11 @@ export default class StatusIcon extends React.PureComponent {
         let className = 'status ' + this.props.className;
 
         if (button) {
-            className = this.props.className;
+            className = this.props.className || '';
```

Comment:
why the fallback? `className` seems to have a default value

Reply:
If I leave out  the fallback here I get this TS error on ```className```:

TS2322: Type 'string | undefined' is not assignable to type 'string'.   Type 'undefined' is not assignable to type 'string'.

The way I have it written here it seems like the TS compiler isn't taking the defaultProps into account and is expecting ```this.props.className``` to possibly be undefined since it's marked as optional in the ```Props``` type.

Another alternative is to remove the ```?```  from the ```className```  property in the ```Props``` type  so that it's just ```className: string;```.  Then the fallback can be removed from and the error goes away.

Actually, If I'm understanding [this section of the TS docs](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#support-for-defaultprops-in-jsx) and this [SO post correctly](https://stackoverflow.com/a/37282264), in TS 3.0+ it's recommended to omit the ```?``` in the ```Props``` type definition for any prop that has a default value supplied in ```defaultProps```.

Any thoughts on this?

Reply:
@enahum @devinbinnie Thoughts on this?  Feel free to merge if all are good.

Reply:
I’m good